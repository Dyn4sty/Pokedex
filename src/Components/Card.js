import React from "react";

const Card = ({ name, url, id }) => {
  const PokeIndex = url.split("/")[url.split("/").length - 2];
  return (
    <div className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img
        alt="pokemon"
        width="150"
        height="150"
        src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${PokeIndex}.png`}
      />
      <div>
        <h2 className="ttc">{name}</h2>
        <p>#{PokeIndex}</p>
      </div>
    </div>
  );
};
export default Card;
