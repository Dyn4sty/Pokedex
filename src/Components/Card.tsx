import React from "react";

interface CardStatelessProps {
  name: string;
  url: string;
};

const Card: React.FunctionComponent<CardStatelessProps> = ({ name, url }) => {
  const PokeIndex: string = url.split("/")[url.split("/").length - 2];
  return (
    <div className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img
        alt="pokemon"
        width="150"
        height="150"
        src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${PokeIndex}.png`}
        loading="lazy"
      />
      <div>
        <h2 className="ttc">{name}</h2>
        <p>#{PokeIndex}</p>
      </div>
    </div>
  );
};
export default Card;
