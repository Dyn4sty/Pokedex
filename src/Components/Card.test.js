import React from 'react';
import { shallow } from 'enzyme';

import Card from './Card';

it('renders the Card Component', () => {
    const mockUrl = "https://pokeapi.co/api/v2/pokemon/?limit=807&offset=0"
    expect(shallow(<Card url={mockUrl} />)).toMatchSnapshot()
})