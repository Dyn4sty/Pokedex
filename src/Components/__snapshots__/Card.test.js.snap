// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card Component 1`] = `ShallowWrapper {}`;
